### PR TITLE
AT2_37: Add in a new page to the SKA developer guide for javascript projects

### PR DESCRIPTION
**Pull request by mark-nicol**
The basic version of the documentation for javascript developers added to the documentation.

### CODE REVIEW

File: src/development/javascript-codeguide.rst [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+Dependencies
+============
+
+* The use  of node and npm is assumed for package management. 
+  All packages and dependencies required to build and run the code should be defined in the ``package.json.``
+  for the project.
+* Avoid relying on any the installation of any global packages to run or build the code. 
+* Take care to differentiate between dependencies and devDependencies when installing packages. The ‘dependencies’ section should only include the dependencies required to run the code. The devDependencies should be used for any packages required to build, test or to deploy the code. 
+* Only install packages for a reason. Adding any new third party dependency should be a team decision, and preferably discussed with the system team.
```
Comment (flyingfrog81) [inline src/development/javascript-codeguide.rst:86]: Great catch, I like it. 
Note that it is in the scope of the system team to create a repository of officially supported binary artefacts (packages, libraries, containers ... )  which will be subject to a change control process.
Once we have ti running we will enforce installation of packages from our trusted repo and thus documentation will need an update, but for now I think it is good as-is. 
thanks,

**Comment by flyingfrog81** [comment]
thanks @mark-nicol ! This is a very useful starting point. 
One aspect we need to cover in the immediate future is what versions of what tools we are using, but this PR looks great to me! Kudos.

File: src/development/javascript-codeguide.rst [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+* Access to git (e.g. Git Bash if on Windows environment (https://gitforwindows.org/)
+* Access to node and the node package manager (npm)
+
+Setting up a new web project
+============================
+
+Each new web project should start by creating a new SKA GitHub project as a fork of the 
+ska-telescope/ska-react-webapp-skeleton project.  
+
+* The readme file will need updated to reflect the nature of the new web project
```
Comment (bravostuzero) [inline src/development/javascript-codeguide.rst:25]: small typo:
* The readme file will need to be updated to reflect the nature of the new web project

**Review by bravostuzero — APPROVED** [review]
Thanks Mark, looks good to me.

**Pull request merged**